### PR TITLE
[IMP] knowledge: revamp onboarding data articles and views

### PR DESCRIPTION
**Pull request by schoffeniels**
PURPOSE:
- This commit slightly modifies the content of the onboarding data
articles: some paragraphs have swapped their position, typos have been
fixed, downloadable files have other names or content, advanced features
paragraph is more detailed.

- It also prevents data articles to reappear when upgrading the module
after that they have been deleted: if one delete a data article, it
should behave the same way as for a regular article and this article
should be deleted forever.

- Add sample data for the favorite and member models, using the data
articles.

- Add no content helpers on the favorite and member views, as well as
on the main "article" view: when no article exists, a smiling face
suggests to create one, when no favorite relation exists, a smiling
face suggests to add an article to the favorites and explains how to
do it, and when no member relation exists, the purpose of creating
one is explained.

These additions have been made with the aim of improving the user
experience.

Task-2852946

### CODE REVIEW

File: addons/knowledge/data/knowledge_data.xml [hunk 1]
```diff
@@ -530,6 +532,8 @@
             </field>
             <field name="parent_id" eval="ref('knowledge_article_sales_and_marketing')" />
             <field name="internal_permission">write</field>
+            <field name="article_member_ids" eval="[(5, 0), (0, 0, {'partner_id': ref('base.main_partner'), 'permission': 'read'}),
```
Comment (tde-banana-odoo) [inline addons/knowledge/data/knowledge_data.xml:535]: Why ? Not sure why we should give member rights on default data... Would keep it as simple as possible, keeping the write internal permission seems sufficient to me.

If you want demo data, should should be done in a demo file.

File: addons/knowledge/data/knowledge_data.xml [hunk 2]
```diff
@@ -145,8 +145,9 @@
             </field>
             <field name="parent_id" eval="False" />
             <field name="internal_permission">write</field>
+            <field name="favorite_ids" eval="[(5,0), (0, 0, {'sequence': 0, 'user_id': ref('base.user_admin')})]"/>
```
Comment (tde-banana-odoo) [inline addons/knowledge/data/knowledge_data.xml:148]: Same remark: to move to demo if necessary, as standard data should not bloat your db. Demo data are there for that purpose.

File: addons/knowledge/data/knowledge_data.xml [hunk 3]
```diff
@@ -185,20 +186,27 @@
             </field>
             <field name="parent_id" eval="False" />
             <field name="internal_permission">write</field>
+            <field name="favorite_ids" eval="[(5, 0), (0, 0, {'sequence': 0, 'user_id': ref('base.user_admin')})]"/>
```
Comment (tde-banana-odoo) [inline addons/knowledge/data/knowledge_data.xml:189]: Same here

File: addons/knowledge/data/article_templates.xml [hunk 4]
```diff
@@ -111,19 +130,40 @@
...
         <div class="o_knowledge_file_extension">pdf</div>
     </div>
     <div class="o_knowledge_toolbar_anchor o_knowledge_toolbar_type_file mt-auto ml-auto" contenteditable="false"><div class="o_knowledge_toolbar" contenteditable="false">
 </div></div>
 </div>
 <br />
+<p>
+    <span style="font-size: 14px;">Want to go <span style="font-style: italic;">even</span> faster? ⚡️
+    <br />
+    Access <span style="font-weight: bolder;">Articles</span> by opening the <font class="text-o-color-1" style="font-weight: bolder;">Command Palette</font> (Ctrl + k/⌘ + k) then search through articles by starting your query with "<span style="font-weight: bolder;">?</span>".
```
Comment (tde-banana-odoo) [inline addons/knowledge/data/article_templates.xml:163]: Could you multi-line ?

File: addons/knowledge/models/res_users.py [hunk 5]
```diff
@@ -39,7 +39,7 @@ def _generate_tutorial_articles(self):
                 'icon': "👋",
                 'internal_permission': 'none',
                 'favorite_ids': [(0, 0, {
-                    'sequence': 0,
+                    'sequence': 1,
```
Comment (tde-banana-odoo) [inline addons/knowledge/models/res_users.py:42]: Why this change ?

File: addons/knowledge/views/knowledge_article_views.xml [hunk 6]
```diff
@@ -192,6 +192,18 @@
                                                                  aria-labelledby="o_article_emoji" />
                                                         </div>
                                                     </div>
+                                                    <!-- No article helper -->
+                                                    <field name="category" invisible="1"/>
+                                                    <div class="o_view_nocontent" attrs="{'invisible': [('create_uid', '!=', False)]}">
```
Comment (tde-banana-odoo) [inline addons/knowledge/views/knowledge_article_views.xml:197]: Shouldn't you be based on id instead ?

**Review by tde-banana-odoo — COMMENTED** [review]
Zbouik zbouik

File: addons/knowledge/models/res_users.py (hunk 5 above)
Comment (schoffeniels) [reply addons/knowledge/models/res_users.py:42]: Since I added sample data in the favourite articles, this tutorial article was not the one opened by default anymore.
I felt like this welcome article should be the one opened by default, even if there are sample favourite articles, so I made this change.
However, since you suggested to move these samples to a demo file, I can revert this modification if it does not matter that the tutorial article is not the one opened by default when there is demo data.

**Comment by tde-banana-odoo** [comment]
Closed in favor of https://github.com/odoo/odoo/pull/93690
